---
title: "Zadanie 7"
output: html_document
date: "2024-11-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
install.packages("Shortread")
library(ShortRead)
```

```{r}
fq1 <- readFastq("C:/Users/homer/Downloads/SRR31368982.fastq")
```
```{r}
qa_result <- qa("C:/Users/homer/Downloads/SRR31368982.fastq")
report(qa_result, dest="C:/Users/homer/Downloads")
```
```{r}
library(Biostrings)

ref_genome <- readDNAStringSet("C:/Users/homer/Downloads/ecoli_genome.fna.gz")

```
```{r}
BiocManager::install("Rsubread")
library(Rsubread)
```
```{r}
buildindex(basename = "ecoli_index", reference ="C:/Users/homer/Downloads/ecoli_genome.fna.gz")
```
```{r}
align(index = "ecoli_index",
      readfile1 = "C:/Users/homer/Downloads/SRR31368982.fastq",
      input_format = "FASTQ",
      output_file = "C:/Users/homer/Downloads/aligned_sample.BAM")
```
```{r}
install.packages("GenomicAlignments")
library(GenomicAlignments)
```


```{r}
aln <- readGAlignments("C:/Users/homer/Downloads/aligned_sample.BAM")
coverage_data <- coverage(aln)
```


```{r}
library(ggplot2)
cov_df <- as.data.frame(coverage_data[[1]])
head("cov_df")
cov_df$position <- as.numeric(rownames(cov_df))


pdf("C:/Users/homer/Downloads/pliku.pdf", width = 8, height = 6)

ggplot(cov_df[1:25000, ], aes(x = position, y = value)) +
  geom_line(color = "green") +
  labs(title = "Pokrycie genomu E. coli",
       x = "Pozycja",
       y = "Liczba zmapowanych odczytÃ³w")
       
dev.off()
```


```{r}
```